### PR TITLE
Kdenlive/mlt: enable glaxnimate to add vectorial animations

### LINKED ISSUE #209923: Mlt glaxnimate missing from Kdenlive
### Describe the bug

When starting KDEnlive from a fresh install/user, a warning is printed:

image

saying that `mediainfo` is missing (so technical info regarding video can't be displayed), and the `Missing MLT module: glaxnimate` needed for Lottie animation (vectorial animations).

Note that that you can still ignore the error, the message being displayed only the first time Kdenlive starts. It can be seen again by going into `Settings / Run config wizzard`

### Steps To Reproduce
```
$ mkdir /tmp/home # or go to «Settings / Run config wizzard»
$ export HOME=/tmp/home
$ nix run github:NixOs/nixpkgs/0c9aadc8eff6daaa5149d2df9e6c49baaf44161c#kdenlive
```

Note that glaxnimate is packaged into nixpkgs since today https://github.com/NixOS/nixpkgs/issues/208707 and the issue was initially reported here https://discourse.nixos.org/t/kdenlive-missing-mlt-module-glaxnimate/24486/4

### Possible fix

The error on mediainfo can easily be fixed by installing `mediainfo`. No idea how to solve glaxnimate, installing glaxnimate is not enough… however after installing glaxnimate the tool appears in Settings / Settings KDEnlive / Environment / Default app / Animation editing`:

image

Note however that in the previous tab `MLT environment` you can see the path to MLT `profile` folder, contained in:
```
$ ls /nix/store/parkimd4avly3ahwsyy869yba9pibp1k-mlt-7.0.1/share/mlt-7/
avformat  core  decklink  frei0r  jackrack  kdenlive  metaschema.yaml  movit  normalize  oldfilm  plus  plusgpl  presets  profiles  qt  resample  rtaudio  sdl  sox  vid.stab  vorbis  xml
```
### Notify maintainers

@turion 

### Related

https://forums.opensuse.org/t/ontbrekende-mlt-module-glaxnimate-vereist-om-lottie-animaties-te-maken/153692/10
the code
https://github.com/KDE/kdenlive/blob/master/src/dialogs/wizard.cpp#L345

### PR DESCRIPTION
**Pull request by tobiasBora**
###### Description of changes

Fix https://github.com/NixOS/nixpkgs/issues/209923

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@turion @goibhniu

### CODE REVIEW

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (turion) [inline pkgs/development/libraries/mlt/qt-5.nix:39]: What happens if you don't do this?

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (turion) [inline pkgs/development/libraries/mlt/qt-5.nix:64]: These two lines are somewhat contradictory: Either you rebuild glaxnimate here, or you use the build input.

Comment (turion) [inline pkgs/development/libraries/mlt/qt-5.nix:64]: Maybe it's possible to build glaxnimate in a different way such that it produces a library? Maybe the glaxnimate derivation can be changed to contain several outputs?

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/qt-5.nix:39]: It fails to find some `*.in.hpp` files downloaded by the submodule.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/qt-5.nix:64]: Could be, but I found no easy way reading at the documentation/cmake file, and I'm not extremely fluent with cmake files nor the internal details of these prolects.

Comment (turion) [reply pkgs/development/libraries/mlt/qt-5.nix:64]: I have next to no idea about cmake, or C/C++ building in general :/ I think we should rather stall this PR here until there is some feedback from the mlt folks, or another idea by someone else from nixpkgs.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/mlt/qt-5.nix:75]:
```suggestion
  ] ++ lib.optional enableGlaxnimate "-DMOD_GLAXNIMATE=ON";
```
that is clear from the line already and I am not sure why we link the issue here which can be closed after this is merged.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (SuperSandro2000) [reply pkgs/development/libraries/mlt/qt-5.nix:64]: if glaxnimate always requires it's buildInputs to be used as a library we should propagate them instead.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (SuperSandro2000) [reply pkgs/development/libraries/mlt/qt-5.nix:39]: Can we symlink our glaxnimate to that location instead?

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/qt-5.nix:64]: Well I use `glaxnimate.buildInputs` only because `mlt` tries to compile glaxnimate from scratch, and I guess it's cleaner to bring all dependencies this way rather than copy pasting the 10 dependencies of glaxnimate.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/qt-5.nix:39]: You mean the final derivation? If yes, no because glaxnimate's output derivation is just a binary + some share files (icons, plugins, python scripts…), while here it tries to use internal functionality of glaxnimate that are not exposed as libraries.

**Comment by tobiasBora** [comment]
So I got an answer by glaxnimate's team: https://gitlab.com/mattbas/glaxnimate/-/issues/545. They say they plan to modify their software to cut it in smaller, reusable libraries, but they had no time to do it so far. So I guess for now it's better to use submodules rather than spending time on writing patches that would eventually be unnecessary.

If you are good with that, I can remove `glaxnimate` from the list of dependencies, squash & modify the commit name once @SuperSandro2000 gives me the appropriate name to give to the commit, and we can merge if everything is good for you.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/qt-5.nix:64]: @turion so I got some feedbacks from MLT: they won't do anything themself but are open to PR. Regarding glaxnimate, they want to refactor the project to provide explicit small reusable libraries, but so far they had no time to do it. So for now I don't think they want to change their build system to include libraries that won't be available anymore after the refactorization.

**Comment by SuperSandro2000** [comment]
> Do I need to write one commit per program?

Yes

> So I guess for now it's better to use submodules rather than spending time on writing patches that would eventually be unnecessary.

If the required work is just installing the so, even if it is done via a cp command and linking against that, then that is preferred. Package collections, like nixpkgs, want to avoid vendoring as much as possible because it makes having a single source for a package with all patches impossible.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (chayleaf) [reply pkgs/development/libraries/mlt/qt-5.nix:39]: can you use `glaxnimate.src` then?

Comment (tobiasBora) [reply pkgs/development/libraries/mlt/qt-5.nix:39]: I have to admit that I don't see much advantage of using `glaxnimate.src` as an upgrade to glaxnimate could easily break the current package, for instance if glaxnimate removes or rename the C functions used in this plugin… (which will certainly arrive as they plan to refactor the code of the library) And I don't see much advantage over fetching the module, anI missing something?

**Comment by HiWhatName** [comment]
Any updates on this?

**Comment by tobiasBora** [comment]
> If the required work is just installing the so, even if it is done via a cp command and linking against that, then that is preferred.

@SuperSandro2000 I understand that Nix tries to limit third party sources, but the software needs the source since glaxnimate provides no library, so I don't think a .so file is even produced (but I did it a while ago so I might remember poorly). I have no idea how to change the whole compilation process in order to avoid this dependency on the source (and I don't have time to spend one week on this), so if you know how to do it feel free to… until then people won't be able to use glaxnimate in KDEnlive.

**Comment by tobiasBora** [comment]
> Where are we stuck currently?

The code works (or at least used to work, I've not tested recently), I am just waiting for @SuperSandro2000 to accept the fact that I rely on a git submodule following what is done upstream. Since glaxnimate provides no API/.so library for now (it might [come later if they find people with good enough cmake expertise](https://gitlab.com/mattbas/glaxnimate/-/issues/545)), I see no other solution.

File: pkgs/development/libraries/mlt/qt-5.nix
Comment (turion) [reply pkgs/development/libraries/mlt/qt-5.nix:39]: I think fetching it as a submodule is fine as long as it isn't packaged better.

**Review by turion — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by turion** [comment]
@tobiasBora see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions for naming conventions for the commits. I think ideally you would split the commit into two, with these names:

```
kdenlive: enable glaxnimate
mlt: enable glaxnimate
```

**Comment by turion** [comment]
@ofborg build kdenlive

**Comment by turion** [comment]
There is something wrong with the build: 

```
building '/nix/store/qw3y9318g1v0i0s8x90dav38ijg4qva6-mlt-7.22.0.drv'...
Running phase: qtPreHook
Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
error: builder for '/nix/store/qw3y9318g1v0i0s8x90dav38ijg4qva6-mlt-7.22.0.drv' failed with exit code 1;
       last 2 log lines:
       > Running phase: qtPreHook
       > Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
```

https://github.com/NixOS/nixpkgs/pull/209941/checks?check_run_id=20524628144

I can reproduce the build error locally.

**Comment by turion** [comment]
It must somehow be related to this part:

```
  ] ++ lib.optionals (qt != null) [
    qt.wrapQtAppsHook
];
```

But it builds on master. I don't understand what change creates the error.

File: pkgs/development/libraries/mlt/default.nix
Comment (turion) [inline pkgs/development/libraries/mlt/default.nix:103]: I think this doesn't quite work as intended anymore. Maybe the issue is that this unconditionally adds `qtbase` and other qt packages, but `mlt` can only deal with them when `qt != null` in the inputs. Maybe throw an error if `enableGlaxnimate = true && qt == null`?

**Comment by tobiasBora** [comment]
Oh interesting… I was also getting this error when compiling mlt alone, but I thought that it was expected since `qt` was not set, thinking that mlt was only compilable with a non-empty qt. Since kdenlive was able to compile `mlt` I thought that mlt was getting sent qt somehow. But now I'm very confused: why is `mlt` failing to build while `ffmpeg = mlt.override {ffmpeg-full}` is not?

**Comment by tobiasBora** [comment]
Oh, ok, now I get why mlt in kdenlive builds fine: in `pkgs/top-level/qt5-packages.nix`, there is 
```
  mlt = pkgs.mlt.override {
    qt = qt5;
  };
```
so KDEnlive is using the one with Qt, explaining why it is not failing to build.

When I have a bit more time, I will try to see if qt is really needed, and if it is needed I will maybe enable glaxnimate only if qt is not null (throwing an error would break the current version, so I'm thinking it might not be the best idea).

**Comment by ShalokShalom** [comment]
Just as a heads up, this is still the case in the unstable build with Qt6 (kdePackages.kdenlive)

**Comment by tobiasBora** [comment]
Ok thanks. I need to take a bit of time to fix this issue. The next month will be rather busy for me, I'll see if I can find some time before but it's definitely in my todo list.

**Comment by tobiasBora** [comment]
This version should work now (let's wait maybe for the tests to finish first). By just specifying the only actually needed library, we get rid of the mlt error (actually, mlt seems to automatically disable glaxnimate support if qt is disabled, without providing any error).

**Comment by tobiasBora** [comment]
Seems like all tests passes, are we good now?

File: pkgs/applications/kde/kdenlive/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/kde/kdenlive/default.nix:92]: Doesn't this need to be done with substituteAll on the patch file?

Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix:92]: Sure, but `substituteAllInPlace` is already part of Kdenlive's derivation as this file is already patched similarly for other binaries.

Comment (SuperSandro2000) [reply pkgs/applications/kde/kdenlive/default.nix:92]: Can we make this explicit? Using substituteAllInPlace with ambient variables is kinda surpising.

Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix:92]: What do you mean? substituteAllInPlace just expects these variables to be in the environment (whether this is good or not is discussed [here](https://github.com/NixOS/nixpkgs/issues/237216), but so far no agreement has been reached), and my goal was just to implement this fix following what kdenlive's package is already doing, as I don't want to risk adding bugs doing some sort of refactoring. Or do you mean to put them in `env = {}` to make it somewhat clearer?

File: pkgs/development/libraries/mlt/default.nix
Comment (chayleaf) [inline pkgs/development/libraries/mlt/default.nix:39]:
```suggestion
```

File: pkgs/applications/kde/kdenlive/default.nix
Comment (chayleaf) [reply pkgs/applications/kde/kdenlive/default.nix:92]: I agree, it would be cleaner to use the substituteAll builder on the patch itself. That said, I don't mind either way.

**Comment by turion** [comment]
@tobiasBora Hi :) do you want to rebase? I'd like to merge it.

**Comment by K900** [comment]
This will only affect the soon-to-be-removed Qt5 version. The Qt6 version is in pkgs/kde/gear/kdenlive.

File: pkgs/applications/kde/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix]: I added an `env` in case you find it clearer, hopefully it will not introduce side effects (but after a quick test it seems to work). Any larger refactoring should be done in a different PR.

**Comment by tobiasBora** [comment]
Argggggggggg We should really follow a policy to accept PR faster, the current system wastes the time and energy of reviewers & maintainers…

**Comment by turion** [comment]
> Argggggggggg We should really follow a policy to accept PR faster, the current system wastes the time and energy of reviewers & maintainers…

Yes, absolutely agree. I only learned about the Qt6 version being a different file today as well.

Now I guess the glaxnimate fix have to be transported to the Qt6 version?

**Comment by tobiasBora** [comment]
Ok, the latest version works with Qt6. Compile with:
```
nix-build -A kdePackages.kdenlive
```

File: pkgs/kde/gear/kdenlive/default.nix
Comment (K900) [inline pkgs/kde/gear/kdenlive/default.nix:17]: Again, does this really need to be conditional?

Comment (tobiasBora) [reply pkgs/kde/gear/kdenlive/default.nix:17]: Sorry, I missed your first message, you are right I can make itnon optional. I also realized (thanks oborg) that my new commit fails to compile flowblade, so wait before merging.

**Comment by tobiasBora** [comment]
Really? You made sure to use the very latest version d27c900cb723f4c6f03941c7c0b33a92e6411032? (Qt6 used to have this bug on a previous commit because I was not compiling mlt for Qt6, but I just tested both Qt5 & 6 with this commit  and both work for me…). If yes, are both versions (Qt5 & 6) failing? If yes, you can maybe try to reset the configuration, maybe it saved some outdated configuration. For this, try to run:
```
rm .config/kdenlive*
```
(or maybe do a backup if you value this config)

**Comment by turion** [comment]
I used the version you mentioned, and tested it by creating a nixos test VM with this config:

```nix
{
    imports = [
      ./common/x11.nix
    ];

    services.xserver.enable = true;
    environment.systemPackages = with pkgs; [
      kdePackages.kdenlive
    ];
  }
```
So the home directory is always pristine.

Running the configuration wizard shows no error message, so something is working :sweat_smile: still, "Create animation" is greyed out:

image

**Comment by tobiasBora** [comment]
I tried to investigate a bit, and I think it is a bug in Kdenlive that occurs only the first time you start it (or if you start it after an order version). I get exactly the first issue (with also "add Title clip" greyed like you, which is also something that should not be greyed) if I do:
```
$ rm ~/.config/kdenlive*
$ ./result/bin/kdenlive
```
but if I just close and restart kdenlive, it works again!

**Comment by turion** [comment]
> but if I just close and restart kdenlive, it works again!

True, can reproduce that! Apart from this weird behaviour (which I can observe both in XFCE and Gnome), it seems to work fine :) well done, fantastic! Thank you in particular for your stamina on this PR!

File: pkgs/kde/gear/kdenlive/default.nix
Comment (turion) [reply pkgs/kde/gear/kdenlive/default.nix:17]: > that my new commit fails to compile flowblade

Is this issue resolved?

**Review by turion — APPROVED** [review]

File: pkgs/kde/gear/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/kde/gear/kdenlive/default.nix:17]: Fixed

**Comment by tobiasBora** [comment]
Great thanks! Are we waiting for something else to merge? If we don't do it now, I'm afraid to have to solve yet another merge conflict in X months ^^'

**Comment by turion** [comment]
There was the one discussion about the optional argument `enableGlaxnimate`, but I think it is not critical.

**Comment by tobiasBora** [comment]
This one is fixed, glaxnimate is now enabled unconditionally in Kdenlive (which makes sense since it is anyway required by kdenlive if we want no error at startup). Thanks!

**Comment by K900** [comment]
Respectfully, what the hell? How did this get merged in the current state? The implementation isn't consistent across Qt versions, none of the maintainers signed off on this, and now we'll have to do a follow-up PR because I don't want to have this in tree in this state.

File: pkgs/applications/kde/kdenlive/default.nix
Comment (K900) [inline pkgs/applications/kde/kdenlive/default.nix:89]: Why are these two patches separate?

File: pkgs/applications/kde/kdenlive/default.nix
Comment (K900) [inline pkgs/applications/kde/kdenlive/default.nix:86]: Comment is outdated now.

File: pkgs/applications/kde/kdenlive/default.nix
Comment (K900) [inline pkgs/applications/kde/kdenlive/default.nix:91]: This comment does not explain what all the other things are for.

File: pkgs/development/libraries/mlt/default.nix
Comment (K900) [inline pkgs/development/libraries/mlt/default.nix:39]: Does this need to be optional here?

File: pkgs/kde/gear/kdenlive/default.nix
Comment (K900) [inline pkgs/kde/gear/kdenlive/default.nix:26]: This does nothing

File: pkgs/kde/gear/kdenlive/default.nix
Comment (K900) [inline pkgs/kde/gear/kdenlive/default.nix:39]: Because this patch is separate (again, why?)

File: pkgs/kde/gear/kdenlive/default.nix
Comment (K900) [inline pkgs/kde/gear/kdenlive/default.nix:54]: And this is still optional (again, why?)

**Comment by tobiasBora** [comment]
> Respectfully, what the hell? How did this get merged in the current state?

Sorry, I did made a mistake in the PR by making glaxnimate only optional for one Qt version, and given that I addressed your last comment I thought my changes would certainly be good enough, but I was certainly a bit too much in a hurry.

And I did not explicitly asked to merge, I just asked what would happen now since originally turion contacted me to just rebase it and I did it ^^' But I'm wondering if it would make sense to create a bot that, when requested, can automatically merge a PR after X days and warn other PR that a PR will be merged soon if it conflicts. This way, it allows a bit of time for the other maintainers to catchup on the changes, without having this annoying "let's wait the answer of XXX", that finally never arrives, and stuck the PR forever until a merging conflict is solved…

Also, even if reaching the perfect code is definitely great, I think we need to find a balance between perfect code and reasonable time lost on improving the code, and compare it with the added value of this cleaning: this can be very time and energy (I mean personal energy but actually also in CPU power ^^) consuming, as any refactoring needs full recompilations, tests, not only from the contributor, but also from all reviewers, and each reviewing round increases the changes for the PR to be abandoned since any tiny change can turn into a massive lost of time. 

> The implementation isn't consistent across Qt versions.

You mean because `glaxnimate` is optional in one qt version and not the other? That's my mistake, I thought I fixed it in the two and cause I was a bit busy I overlooked it.

> none of the maintainers signed off on this

What do you mean? @turion is a maintainer of kdenlive, and except from you, no one got active since @turion contacted me.

> and now we'll have to do a follow-up PR because I don't want to have this in tree in this state.

Done in https://github.com/NixOS/nixpkgs/pull/324552

File: pkgs/kde/gear/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/kde/gear/kdenlive/default.nix:54]: sorry, my mistake, fixed in https://github.com/NixOS/nixpkgs/pull/324552

File: pkgs/kde/gear/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/kde/gear/kdenlive/default.nix:39]: fixed in https://github.com/NixOS/nixpkgs/pull/324552

File: pkgs/kde/gear/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/kde/gear/kdenlive/default.nix:26]: fixed in https://github.com/NixOS/nixpkgs/pull/324552

File: pkgs/development/libraries/mlt/default.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/default.nix:39]: Since the library creates a compilation flag to enable this or not, I think it makes sense to make it optional here… But to be honest I don't care so much.

File: pkgs/applications/kde/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix:91]: removed in https://github.com/NixOS/nixpkgs/pull/324552

File: pkgs/applications/kde/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix:86]: what do you mean? I never wrote this comment I think ^^

Comment (K900) [reply pkgs/applications/kde/kdenlive/default.nix:86]: Yes, but you changed what the patch does, so the comment needs updating as well.

File: pkgs/development/libraries/mlt/default.nix
Comment (K900) [reply pkgs/development/libraries/mlt/default.nix:39]: Not every optional dependency should be a compilation flag, or we'll create an unmaintainable combinatoric explosion of possible flags.

**Comment by turion** [comment]
I agree with all of the above.

@K900 Apologies that the state of the PR didn't meet your expectations. It is over a year old, has been rebased, approved, commented on, people dropped by asking for changes and then were unresponsive for a long time. I found it frustrating that while tobiasBora has put effort in this, been patient, and created value for the community, there was always something that held the PR back. So I decided to go forward in an imperfect state. Yes, I could have waited for your extensive and detailed feedback that certainly improves the code quality. But I've also seen enough PRs that slowly wither because there is review after review burning out the contributor. I perceived losing tobiasBora here, and I didn't want to. 

> none of the maintainers signed off on this

I'm the sole "maintainer" of the old kdenlive (I put quotes here because I'm not proficient enough in Qt to do any nontrivial maintenance like tobiasBora has done). I have no idea how to identify the correct maintainer of the plasma 6 version, of which I was completely unaware until recently. It seems that a group of people is now maintaining all of the kdePackages at the same time? Should I have pinged all of them?

**Comment by K900** [comment]
Generally, the KDE team maintains all KDE 6 packages as a baseline. At least the committers on the team are pinged automatically by CODEOWNERS.

File: pkgs/applications/kde/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix:89]: Technically, it was mostly to save me 2h of debugging & testing & compiling while I was quite busy for other stuff. I tried before to merge the two patches manually, but it turned out that for a reason I can't explain my patch was not applied (but I did not realized it directly), making me lose quite some time, first to realize that my patch was not applied, and then to find a way to combine these patches appropriately + burn some CPU power to compile both versions and redo a session of tests etc… And I thought that this small additional patch was not such a bad thing compared with the time it would save me ^^

File: pkgs/development/libraries/mlt/default.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/default.nix:39]: Alright, I don't mind to remove it then. I saw that you plan to work on it now, should I let you do it?

Comment (K900) [reply pkgs/development/libraries/mlt/default.nix:39]: I'll remove it in the 24.05.2 PR.

File: pkgs/applications/kde/kdenlive/default.nix
Comment (K900) [reply pkgs/applications/kde/kdenlive/default.nix:89]: So you've chosen to save your time now instead of saving everyone's time later?

File: pkgs/development/libraries/mlt/default.nix
Comment (tobiasBora) [reply pkgs/development/libraries/mlt/default.nix:39]: Ok thanks a lot!

File: pkgs/applications/kde/kdenlive/default.nix
Comment (tobiasBora) [reply pkgs/applications/kde/kdenlive/default.nix:89]: Well, I'm not so sure if this will make anyone lose 2h later since it does not harm much to have two files instead of 1. But don't get me wrong, I do get your point, especially for important architecture decisions, where I'm the first one to care about having a clean base on which to build on… this was just my tradeoff between answering quickly to turion's request without getting late on the other things I need to do. Obviously this had not the effect I was hopping for, so I know what I'm gonna do tonight :-P

**Pull request merged**
